app/src/main/java/com/example/gradeassist/service/IQuestionService.kt

```diff
@@ -0,0 +1,36 @@
+package com.example.gradeassist.service
+
+import androidx.lifecycle.MutableLiveData
+import com.example.gradeassist.RetrofitClientInstance
+import com.example.gradeassist.dao.IQuestionDAO
+import com.example.gradeassist.dto.Question
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+interface IQuestionService {
+
+    fun fetchQuestionById(questionId: Int) : MutableLiveData<Question> {
+        var _question = MutableLiveData<Question> ()
+        val service = RetrofitClientInstance.retrofitInstance?.create(IQuestionDAO::class.java)
+        service?.getQuestionById(questionId)
+        val call = service?.getQuestionById(questionId)
+        call?.enqueue(object: Callback<Question> {
+            override fun onResponse(
+                call: Call<Question>,
+                response: Response<Question>
+            ) {
+                _question.value = response.body()
+            }
+
+            override fun onFailure(call: Call<Question>, t: Throwable) {
+                //TODO finish onFailure Function
+            }
+
+        })
+
+        return _question
+
+
+    }
+}
```

Comment:
Added a new interface for service (IQuestionService) with a function that fetches question by id with retrofit to parse JSON

Reply:
Just curious... why an interface, not a class?

Reply:
I was curious as well when i was going through the review, the project already had an interface for their IScoreService and not a class. I wanted to create a class for it but decided to follow their design pattern and created an interface.